nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1025,6 +1029,17 @@ public static boolean isStringTypeCompatible(final Object value) {
         return value != null;
     }
 
+    public static boolean isEnumTypeCompatible(final Object value, final EnumDataType enumType) {
+        return enumType.getEnums() != null && enumType.getEnums().contains(value);
+    }
+
+    private static Object toEnum(Object value, EnumDataType dataType, String fieldName) {
+        if(dataType.getEnums() != null && dataType.getEnums().contains(value)) {
```

Comment:
Are you assuming only conversion from strings here? Like not even allowing people to convert booleans to TRUE and FALSE?

Reply:
I assumed that we can only have strings: http://avro.apache.org/docs/current/spec.html#Enums

Reply:
Yeah, I know the spec only allows strings, but I was wondering if wanted to allow boolean values to be converted to strings here.

Reply:
We should be very careful here - there is no spec. We absolutely do not - and should not - follow the Avro spec. We should do what makes sense for us. Personally I am not a fan of introducing Enums at all. I very intentionally didn't add those in as a data type when we created the initial set.  More on that in a general comment below :)